src/charm.py

```diff
@@ -893,14 +900,21 @@ def _on_flush_runners_action(self, event: ActionEvent) -> None:
             runner_scaler = self._get_runner_scaler(state)
             flushed = runner_scaler.flush(flush_mode=FlushMode.FLUSH_IDLE)
             logger.info("Flushed %s runners", flushed)
-            delta = runner_scaler.reconcile(state.runner_config.virtual_machines)
+            self.unit.status = MaintenanceStatus("Reconciling runners")
+            try:
+                delta = runner_scaler.reconcile(state.runner_config.virtual_machines)
+            except ReconcileError:
+                logger.exception("Failed to reconcile runners")
+                event.fail("Failed to reconcile runners")
```

Comment:
Same comments as below, with the addition that the maintenance status will keep the "Reconciling runners" message